capstone/scripts/convert_s3.py

```diff
@@ -89,123 +89,135 @@ def export_cases_to_s3(redacted: bool, reporter_id: str) -> None:
-                key = f"{volume_prefix}/CasesMetadata.jsonl"
-            else:
-                key = f"{volume_prefix}/Cases.jsonl"
-
-            # store the serialized case data in tempfile
-            with tempfile.NamedTemporaryFile() as file:
-                for item in cases_search.scan():
-                    # pass case in to add additional data to the CaseDocument
-                    case = CaseMetadata.objects.get(pk=item["_source"]["id"])
-                    serializer = vars["serializer"](
-                        item["_source"],
-                        context={
-                            "request": vars["fake_request"],
-                            "first_page_order": case.first_page_order,
-                            "last_page_order": case.last_page_order,
-                        },
-                    )
-                    file.write(json.dumps(serializer.data).encode("utf-8") + b"\n")
-                file.flush()
-                # not closing with loop so I can continue using file for upload
-                hash_and_upload(file, dest_bucket, key, "application/jsonl")
+    cases_by_id = {case.pk: case for case in cases}
+    frontend_url = cases[0].frontend_url
 
-        # copies each volume PDF to new location if it doesn't already exist
-        copy_volume_pdf(volume, volume_prefix, dest_bucket, redacted)
+    if len(cases) == 0:
+        print("WARNING: Volume '{}' contains NO CASES.".format(volume.barcode))
+        return
 
-        # export each case in the volume
-        for case in cases:
-            export_single_case(case, f"{volume_prefix}/case", dest_bucket)
+    volume_prefix = "/".join(frontend_url.rsplit("/")[1:3])
 
+    put_volume_metadata(dest_bucket, volume, volume_prefix)
 
-def export_single_case(case: object, case_prefix: str, bucket: str) -> None:
-    """
-    Put full text plus metadata of each case in reporterX/volumeX/case/caseX.jsonl.
-    """
-    # find name of the paragraph that is tied to the case order from the volume PDF
+    cases_key = f"{volume_prefix}/Cases/"
+
+    # fetch existing files to compare to what we have
+    s3_contents_hash = fetch_s3_files(dest_bucket, cases_key)
 
-    # set up vars for text format
-    vars = {
-        "serializer": ConvertNoLoginCaseDocumentSerializer,
-        "query_params": {"body_format": "text"},
-    }
     # fake Request object used for serializing case with DRF's serializer
     vars["fake_request"] = namedtuple("Request", ["query_params", "accepted_renderer"])(
         query_params=vars["query_params"],
         accepted_renderer=None,
     )
+    # fake Request object used for serializing cases with DRF's serializer
+    vars["fake_request"] = namedtuple("Request", ["query_params", "accepted_renderer"])(
+        query_params={"body_format": "text"},
+        accepted_renderer=None,
+    )
+
+    # create a metadata contents string to append case metadata content
+    metadata_contents = b""
```

Comment:
As a tiny nitpick, I think python is smoother if you keep everything internally in strings, and just do `.encode("utf-8")` right at the end when you call `hashlib.sha256` or `file.write`. In this case you can also do `tempfile.NamedTemporaryFile(mode='wt')` to specify opening the file in text mode, and then I don't think you'll need `.encode` at all.